keras/backend/tensorflow/distribution_lib.py

```diff
@@ -35,9 +35,21 @@ def list_devices(device_type=None):
     ]
 
 
-def distribute_value(value, tensor_layout):
-    # TODO
-    pass
+def distribute_variable(value, tensor_layout):
```

Comment:
This is a public API, so it should be implemented with the same name/signature across all backends. If we make it variable-specific for TF, then it should also be variable-specific for JAX. Maybe this is what we should do -- have a function for tensor sharding (which we would use for data sharding and for intermediate tensor sharding) and another for variable sharding. Waiting for input from @qlzh727 for this.

Reply:
yea, since jax doesn't really distinguish between tensor and variable, I think it might make sense to have distribute_tensor and distribute_variable for both backend, and JAX backend will have the same implementation.

Reply:
Let's do that -- also commented on this on the `sharding_constraint` PR.

Reply:
Sounds good. Created two separate functions as suggested